lib/bundlex/toolchain/common/unix/os_deps.ex

```diff
@@ -0,0 +1,175 @@
+defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
+  @moduledoc false
+
+  require Logger
+  alias Bundlex.Output
+
+  @precompiled_path "#{Mix.Project.build_path()}/bundlex_precompiled/"
+
+  @spec get_flags(Bundlex.Native.t(), atom()) :: String.t()
+  def get_flags(native, flags_type) do
+    native.os_deps
+    |> Enum.flat_map(fn
+      {:pkg_config, lib_names} ->
+        get_flags_for_pkg_config(lib_names, flags_type, native.app)
+
+      {precompiled_dependency_path, lib_names} ->
+        get_flags_for_precompiled(
+          {precompiled_dependency_path, lib_names},
+          flags_type
+        )
+    end)
+    |> Enum.uniq()
+    |> Enum.join(" ")
+  end
+
+  defp parse_os_deps(os_deps) do
+    Enum.map(os_deps, fn
+      {precompiled_dependency, lib_name_or_names} ->
+        lib_names = Bunch.listify(lib_name_or_names)
+        {precompiled_dependency, lib_names}
+
+      lib_name ->
+        {:pkg_config, [lib_name]}
+    end)
+  end
+
+  defp get_flags_for_precompiled(
+         {{_precompiled_dependency_url, precompiled_dependency_path}, lib_names},
+         flags_type
+       ) do
+    case flags_type do
+      :libs ->
+        full_packages_library_path = Path.absname("#{precompiled_dependency_path}/lib")
+
+        [
+          "-L#{full_packages_library_path}",
+          "-Wl,-rpath,#{full_packages_library_path}"
+        ] ++
+          Enum.map(lib_names, &"-l#{remove_lib_prefix(&1)}")
+
+      :cflags ->
+        full_include_path = Path.absname("#{precompiled_dependency_path}/include")
+
+        ["-I#{full_include_path}"]
+
+      true ->
+        raise "Unknown flag type: #{inspect(flags_type)}"
+    end
+  end
+
+  defp get_flags_for_pkg_config(lib_names, options, app) do
+    options = options |> Bunch.listify() |> Enum.map(&"--#{&1}")
+    do_get_flags_for_pkg_config(lib_names, options, app)
+  end
+
+  defp do_get_flags_for_pkg_config([], _options, _app), do: ""
```

Comment:
Is this case even possible? Take a look at line 33, first argument of this function will always match `[single_lib_name]`

Reply:
I guess it would be called in case somebody passed the following configuration:
```
os_deps: [{:pkg_config, []}]
```